lib/_http_agent.js

```diff
@@ -123,6 +123,14 @@ Agent.prototype.addRequest = function(req, options) {
   options = util._extend({}, options);
   options = util._extend(options, this.options);
 
+  if (!options.servername) {
+    options.servername = options.host;
+    const hostHeader = req.getHeader('host');
+    if (hostHeader) {
+      options.servername = hostHeader.replace(/:.*$/, '');
```

Comment:
I left the `hostHeader` cleanup intentionally unmodified to match the original feature implemented PR #1110

I did not want to fix two unrelated issues in single PR dedicated to memory leak.